src/algebra/fields/fp.tcc

```diff
@@ -197,11 +197,11 @@ Fp_model<n,modulus>::Fp_model(const long x, const bool is_unsigned)
 {
     if (is_unsigned || x >= 0)
     {
-        this->mont_repr.data[0] = x;
+        this->mont_repr.data[0] = (mp_limb_t)x;
```

Comment:
This is because ``mp_limb_t`` is unsigned, and it is intentional to cast negative ``x`` to unsigned when ``is_unsigned`` is true, I suppose. I don't like this API but ok.

Perhaps add ``BOOST_STATIC_ASSERT(sizeof(mp_limb_t) >= sizeof(long));``.